docs/documentation/cn/developer-guide/debug-tool.md

```diff
@@ -0,0 +1,247 @@
+
+```
+BUILD_TYPE=ASAN ./build.sh
+```
+
+执行编译生成的二进制文件，当检测工具发现有异常访问时，就会立即退出，并将非法访问的堆栈输出在be.out中。对于ASAN的输出与LSAN是一样的分析方法。这里我们也主动注入一个地址访问错误，来展示下具体的内容输出。我们仍然在`StorageEngine`的`open`函数中注入一段非法内存访问，具体的错误代码如下
+
+```
+    char* invalid_buf = new char[1024];
+    for (int i = 0; i < 1025; ++i) {
+        invalid_buf[i] = i;
+    }
+    LOG(INFO) << invalid_buf;
+```
+
+然后我们就会在be.out中获得如下的输出
+
+```
+=================================================================
+==23284==ERROR: AddressSanitizer: heap-buffer-overflow on address 0x61900008bf80 at pc 0x00000129f56a bp 0x7fff546eed90 sp 0x7fff546eed88
+WRITE of size 1 at 0x61900008bf80 thread T0
+    #0 0x129f569 in doris::StorageEngine::open(doris::EngineOptions const&, doris::StorageEngine**) /home/ssd0/zc/palo/doris/core/be/src/olap/storage_engine.cpp:106
+    #1 0xe2c1e3 in main /home/ssd0/zc/palo/doris/core/be/src/service/doris_main.cpp:159
+    #2 0x7fa5580fbbd4 in __libc_start_main (/opt/compiler/gcc-4.8.2/lib64/libc.so.6+0x21bd4)
+    #3 0xd30794  (/home/ssd0/zc/palo/doris/core/output3/be/lib/palo_be+0xd30794)
+
+0x61900008bf80 is located 0 bytes to the right of 1024-byte region [0x61900008bb80,0x61900008bf80)
+allocated by thread T0 here:
+    #0 0xdeb040 in operator new[](unsigned long) ../../../../gcc-7.3.0/libsanitizer/asan/asan_new_delete.cc:82
+    #1 0x129f50d in doris::StorageEngine::open(doris::EngineOptions const&, doris::StorageEngine**) /home/ssd0/zc/palo/doris/core/be/src/olap/storage_engine.cpp:104
+    #2 0xe2c1e3 in main /home/ssd0/zc/palo/doris/core/be/src/service/doris_main.cpp:159
+    #3 0x7fa5580fbbd4 in __libc_start_main (/opt/compiler/gcc-4.8.2/lib64/libc.so.6+0x21bd4)
+
+SUMMARY: AddressSanitizer: heap-buffer-overflow /home/ssd0/zc/palo/doris/core/be/src/olap/storage_engine.cpp:106 in doris::StorageEngine::open(doris::EngineOptions const&, doris::StorageEngine**)
+```
+
+从这段信息中该可以看到在`0x61900008bf80`这个地址我们尝试去写一个字节，但是这个地址是非法的。我们也可以看到 `[0x61900008bb80,0x61900008bf80)`这个地址的申请堆栈。
+
+**注意：开启这个选项是要影响程序的执行性能的，请慎重对线上的实例开启**
+
+**注意：如果开启了LSAN开关的话，tcmalloc就会被自动关闭**
+
+另外，如果be.out中输出了堆栈信息，但是并没有函数符号，那么这个时候需要我们手动的处理下才能获得可读的堆栈信息。具体的处理方法需要借助一个脚本来解析ASAN的输出。这个时候我们需要使用[asan_symbolize](https://llvm.org/svn/llvm-project/compiler-rt/trunk/lib/asan/scripts/asan_symbolize.py)来帮忙解析下。具体的使用方式如下：
+
+```
+cat be.out | python asan_symbolize.py | c++filt
+```
+
+通过上述的命令，我们就能够获得可读的堆栈信息了。
+
+## CPU
+
+当系统的CPU Idle很低的时候，说明系统的CPU已经成为了主要瓶颈，这个时候就需要分析一下当前的CPU使用情况。对于Doris的BE可以有如下两种方式来分析Doris的CPU瓶颈。
+
+### pprof
+
+由于Doris内部已经集成了并兼容了GPerf的REST接口，那么用户可以通过`pprof`工具来分析远程的Doris BE。具体的使用方式如下：
+
+```
+pprof --svg --seconds=60 http://be_host:be_webport/pprof/profile > be.svg 
+```
+
+这样就能够生成一张BE执行的CPU消耗图。
+
+![CPU Pprof](../../../resources/images/cpu-pprof-demo.png)
+
+### perf + flamegragh
+
+这个是相当通用的一种CPU分析方式，相比于`pprof`，这中方式必须要求能够登陆到分析对象的物理机上。但是相比于pprof只能定时采点，perf是能够通过不同的事件来完成堆栈信息采集的。具体的的使用方式如下：
+
+```
+ perf record -g -p be_pid -- sleep 60
```

Comment:
```suggestion
perf record -g -p be_pid -- sleep 60
```

Reply:
Is it `-- sleep 60` or `--sleep 60`(without space between `--` and `sleep`)

Reply:
It is `-- sleep 60`, which sleep is another shell command.